ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/replacehook.js

```diff
@@ -131,26 +131,23 @@
 
                 if (!(keyword.length > 0) || isKeywordFound) {
                     if ((!cannotBeReplacedWith.includes(componentType))
-                        && typeMap[editableType] === typeMap[componentType]
```

Comment:
I would want to move this code to cq-guides, can you create a tracking ticket for this. To me, this logic should not be part of core component, this is not something which forms developers would like to customize ?

Reply:
> This is just an implementation detail for replace hook that may /may not be customised but it doesn't matters to us.

Sorry I beg to differ here, I would want to keep the core component code clean. We have some API's already exposed in cq-guides which is consumed by core components. Core component should only contain code which is meaningful for forms developer to read, else it adds complexity. Replace is an edit action and forms core component developer should just use it (as API's)